modules/uhttp/client/client_filters.go

```diff
@@ -81,6 +82,20 @@ func tracingFilter(
 	return resp, err
 }
 
+func authenticationFilter(
+	ctx fx.Context, req *http.Request, next BasicClient,
+) (resp *http.Response, err error) {
+	authClient := uauth.Instance()
+	authctx, err := authClient.Authenticate(ctx)
+	if err != nil {
+		ctx.Logger().Error(uauth.ErrAuthentication, "error", err)
+		return nil, err
+	}
+	return next.Do(&fxcontext.Context{
+		Context: authctx,
+	}, req)
```

Comment:
nit: This reads a little weird to me. Alternative:
next.Do(
    &fxcontext.Context{Context: authctx},
    req
)